lib/http.js

```diff
@@ -81,17 +81,55 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
     cors(),
     // FIXME: add validation
     //validate({bodySchema: updateMeterBody}),
-    _createAuthorizeMiddleware(),
+    _addMeterRecordToRequest(),
+    _createAuthorizeMiddleware({
+      getExpectedTarget({req}) {
+        // only support update meter config URL as expected target; controller
+        // is the controller of the meter
+        const baseUrl = `https://${req.get('host')}${routes.basePath}`;
+        const expectedTarget =
+          `${baseUrl}/${encodeURIComponent(req.params.meterId)}`;
+        return {expectedTarget};
+      },
+      getRootController({req}) {
+        // root controller for update meter config is always the controller of
+        // the meter
+        const {meter: {controller}} = req.meterRecord;
+        return controller;
+      }
+    }),
     asyncHandler(async (req, res) => {
       let {body: meter} = req;
 
       // update meter according to handler result
       ({meter} = await HANDLERS.update({meter}));
```

Comment:
Maybe we pass in the existing meter record that will be updated so that the custom handler can make decisions (for example, maybe a `controller` change will be illegal if the meter doesn't include a billing zcap that can be checked or something like that?).

```suggestion
      ({meter} = await HANDLERS.update({meter, meterRecord: req.meterRecord}));
```

Reply:
Can we name it `currentMeterRecord` instead and follow a similar pattern used here: https://vuejs.org/v2/api/#watch?